### DISCUSSION TITLE
Reconfiguring an existing Ponder project.

### DISCUSSION THREAD

**Discussion by salieflewis** [ponder-sh/ponder#331] [Question]
I'm using Ponder as a testing environment against a moving smart contract, aka the contract's ABI is changing as we move things around, but the events are staying the same!

I thought it would be as simple as changing the contract address in my `ponder.config.ts` as well as updating the abi in `abis/contract.json` but it doesn't seem to be that simple. In essence Ponder is aware of new events, but the logging capabilities do not persist.

I'm curious if there is a way to kind of reconfig the project given an updated contract without having to go through the whole CLI process and create a new app. Or is there another thing I'm missing that is necessary to change?

**Comment by typedarray** [comment]
I'm not quite sure what behavior you're seeing when you say

> Ponder is aware of new events, but the logging capabilities do not persist.

That said - we're very keen to natively supported Foundry + Ponder integration such that when you save changes to your Foundry project, it'll reload Ponder automatically and behave as you expect.

**Reply by salieflewis** [reply]
That would certainly solve some issues! Essentially what I'm getting at is that I don't want to have to create a whole new ponder project every time I change the address of the contract I'm working with and its associated ABI. If I encounter this again I can try to document what I'm experiencing in the context of a loom video or something.

**Comment by shrugs** [comment]
@salieflewis it might be that you can delete the cache directory `.ponder` after you restart foundry to solve this dx issue
